source/extensions/config_subscription/grpc/grpc_mux_failover.h

```diff
@@ -191,21 +195,19 @@ class GrpcMuxFailover : public GrpcStreamInterface<RequestType, ResponseType>,
       // considering the primary source as available.
       // Calling the onStreamEstablished() callback on the GrpcMux object will
       // trigger the GrpcMux to start sending requests.
-      ASSERT(parent_.connecting_to_primary_ && !parent_.connecting_to_failover_ &&
-             (parent_.connected_to_ == ConnectedState::None));
+      ASSERT(parent_.connection_state_ == ConnectionState::ConnectingToPrimary);
       parent_.grpc_mux_callbacks_.onStreamEstablished();
     }
 
     void onEstablishmentFailure() override {
       // This will be called when the primary stream fails to establish a connection, or after the
       // connection was closed.
-      ASSERT(parent_.connectingToOrConnectedToPrimary() &&
-             !parent_.connectingToOrConnectedToFailover());
+      ASSERT(parent_.connectingToOrConnectedToPrimary());
       // If there's no failover supported, this will just be a pass-through
       // callback.
       if (parent_.failover_grpc_stream_ != nullptr) {
-        if (parent_.connecting_to_primary_ &&
-            (parent_.ever_connected_to_ != ConnectedState::Primary)) {
+        if (parent_.connection_state_ == ConnectionState::ConnectingToPrimary &&
+            !parent_.ever_connected_to_primary_) {
```

Comment:
Is there an opportuntiy to scatter some more ASSERTs with this variable around?

Reply:
Done